hydroflow/src/util/connection.rs

```diff
@@ -0,0 +1,99 @@
+use std::{path::PathBuf, pin::Pin};
+
+use futures::{Sink, Stream};
+use serde::{Deserialize, Serialize};
+use tokio::net::{TcpListener, TcpStream, UnixListener, UnixStream};
+use tokio_util::codec::LinesCodecError;
+
+use super::{tcp_lines, unix_lines};
+
+/// Describes a medium through which two HydroFlow services can communicate.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum ConnectionPipe {
+    UnixSocket(PathBuf),
+    TcpPort(String, u16),
```

Comment:
Maybe more principled to use [std](https://doc.rust-lang.org/std/index.html)::[net](https://doc.rust-lang.org/std/net/index.html)::[SocketAddr](https://doc.rust-lang.org/std/net/enum.SocketAddr.html#) here?